clang/lib/Format/ContinuationIndenter.cpp

```diff
@@ -703,7 +703,9 @@ void ContinuationIndenter::addTokenOnCurrentLine(LineState &State, bool DryRun,
 
   if (Current.is(tok::equal) &&
       (State.Line->First->is(tok::kw_for) || Current.NestingLevel == 0) &&
-      CurrentState.VariablePos == 0) {
+      CurrentState.VariablePos == 0 &&
+      (!Previous.Previous ||
+       Previous.Previous->isNot(TT_DesignatedInitializerPeriod))) {
```

Comment:
Might need to search further back if the element being initialized is an array.
`  struct { int x[2]; } a[] = { [0].x = {1,1}, [1].x[0] = 42};`

Reply:
not a for loop, but similar code should be possible in a for loop initialization

Reply:
maybe we handle that in seperate issue?

Reply:
yes, see if anyone complains anyway.